setup.cfg

```diff
@@ -0,0 +1,35 @@
+[metadata]
+name = matrix-python-common
+description = "Common utilities for Synapse, Sydent and Sygnal"
+version = 0.0.0
+
+
+[options]
+python_requires = >= 3.7
+install_requires =
+  attrs
+
+
+[options.extras_require]
+dev =
+  # for tests
+  tox
+  twisted
```

Comment:
Do we want a dependency on `twisted` for this? It feels a bit heavyweight for a utility library.

Reply:
There's code shared between Synapse and Sydent that I'd hoped to pull into here (#2, #3) which extends or interfaces with twisted. Maybe they'd be better off in a separate library?

Reply:
Can we do some magic to make twisted an optional dependency?

Reply:
extra_requires I think? https://setuptools.pypa.io/en/latest/userguide/declarative_config.html

Reply:
Extras are the way to go, here. There's a really excellent blog post discussing how someone used this approach (with some creative nuance) for a scientific computing utility library... let me see if I can dig that up (don't block on this beyond Monday; I'll either find it tonight or I won't be able to find it at all)

Reply:
Not finding it. If I recall correctly, it did something simple but clever around checking whether optional libraries were available before exposing functions / submodules which relied on them... Oh well.

Reply:
Let's leave it as it is. It's called "Common utilities for Synapse, Sydent and Sygnal" and all three of those projects use twisted. (IMO)